lib/src/main/java/org/esbtools/lightbluenotificationhook/NotificationHook.java

```diff
@@ -110,60 +134,95 @@ private HookResult processSingleHookDoc(Projector watchProjector, Projector incl
         return HookResult.aborted();
     }
 
-    private boolean watchedFieldsHaveChanged(JsonDoc preDoc, JsonDoc postDoc,
-            Projector watchProjector) {
+    /**
+     * Determines if any of the watched fields have changed. If
+     * arrayOrderingSignificant is true, also checks if array
+     * orderings also changed.
+     */
+    private boolean watchedFieldsHaveChanged(EntityMetadata metadata,
+                                             JsonDoc preDoc,
+                                             JsonDoc postDoc,
+                                             Projector watchProjector,
+                                             boolean arrayOrderingSignificant) {
         JsonDoc watchedPostDoc = watchProjector.project(postDoc, jsonNodeFactory);
         JsonDoc watchedPreDoc = preDoc == null
-                ? new JsonDoc(jsonNodeFactory.nullNode())
-                : watchProjector.project(preDoc, jsonNodeFactory);
+            ? new JsonDoc(jsonNodeFactory.objectNode())
+            : watchProjector.project(preDoc, jsonNodeFactory);
 
-        return !watchedPostDoc.getRoot().equals(watchedPreDoc.getRoot());
+        // Compute diff
+        JsonCompare cmp=metadata.getDocComparator();
+        try {
+            DocComparator.Difference diff=cmp.compareNodes(watchedPreDoc.getRoot(),watchedPostDoc.getRoot());
+            LOGGER.debug("Diff: {}",diff);
+            if(!diff.same()) {
+                if(diff.getNumChangedFields()>0) {
```

Comment:
When are two docs different but number of changed fields is 0? Not sure I understand what the true meaning of a "changed field" is in this API, and how that relates to caring about array order changes.